source/postprocess/visualization/dynamic_topography.cc

```diff
@@ -75,21 +77,36 @@ namespace aspect
           if (cell->is_locally_owned())
             if (cell->at_boundary())
               {
-                // see if the cell is at the *top* boundary, not just any boundary
-                unsigned int top_face_idx = numbers::invalid_unsigned_int;
-                {
-                  for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
-                    if (cell->at_boundary(f) && this->get_geometry_model().depth (cell->face(f)->center()) < cell->face(f)->minimum_vertex_distance()/3)
+                // see if the cell is at the *top* or *bottom* boundary, not just any boundary
+                unsigned int face_idx = numbers::invalid_unsigned_int;
+                bool at_upper_surface;
```

Comment:
Actually I think at_upper_surface might make more sense because then true has a clear meaning, which isn't the case if the name is top_or_bottom_boundary_face. But I agree that I should add a comment that false corresponds to the lower boundary. On it!